src/wallet/rpc/wallet.cpp

```diff
@@ -738,16 +741,27 @@ static RPCHelpMan migratewallet()
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
         {
-            std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
-            if (!wallet) return NullUniValue;
-
-            if (wallet->IsCrypted()) {
-                throw JSONRPCError(RPC_WALLET_WRONG_ENC_STATE, "Error: migratewallet on encrypted wallets is currently unsupported.");
+            std::string wallet_name;
+            if (GetWalletNameFromJSONRPCRequest(request, wallet_name)) {
+                if (!(request.params[0].isNull() || request.params[0].get_str() == wallet_name)) {
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, "RPC endpoint wallet and wallet_name parameter specify different wallets");
+                }
+            } else {
+                if (request.params[0].isNull()) {
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, "Either RPC endpoint wallet or wallet_name parameter must be provided");
+                }
+                wallet_name = request.params[0].get_str();
             }
 
-            WalletContext& context = EnsureWalletContext(request.context);
+            // Note that the walletpassphrase is stored in request.params[1] which is not mlock()ed
+            SecureString wallet_pass;
+            wallet_pass.reserve(100);
+            // TODO: get rid of this .c_str() by implementing SecureString::operator=(std::string)
+            // Alternately, find a way to make request.params[0] mlock()'d to begin with.
+            wallet_pass = request.params[1].isNull() ? "" : request.params[1].get_str().c_str();
```

Comment:
Possibly for another PR (and possibly a dumb question), but why can't we do 

```c++
if (!request.params[1].isNull()) wallet_pass.assign(request.params[1].get_str());
```

rather than use `.c_str()` (and in the other existing places)? This is supported since C++17.

Reply:
I've updated the passphrase handling to match what #27068 does